### PR TITLE
add selector spread plugin

### LINKED ISSUE #2503: The lister can't list resource in predicates and nodeorder plugins
**What happened**:
The informers for namespaces resource are not registered to the informer factory, then in `predicates` or `nodeorder` plugins, the namespace lister used can't get any resources：
`E0922 10:33:55.843052       8 plugin.go:143] "getting namespace, assuming empty set of namespace labels" err="namespace \"default\" not found" namespace="default"`

**What you expected to happen**:
the lister can list resources

**How to reproduce it (as minimally and precisely as possible)**:
deploy any workload and use interpodaffinity plugin

**Anything else we need to know?**:

**Environment**:1.6.1
- Volcano Version:1.6.1
- Kubernetes version (use `kubectl version`):1.19
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by elinx**
also fix #2503

### CODE REVIEW

File: pkg/scheduler/plugins/nodeorder/nodeorder.go [hunk 1]
```diff
@@ -145,6 +148,9 @@ func calculateWeight(args framework.Arguments) priorityWeight {
 	// Checks whether imagelocality.weight is provided or not, if given, modifies the value in weight struct.
 	args.GetInt(&weight.imageLocalityWeight, ImageLocalityWeight)
 
+	// Checks whether selectorspread.weight is provided or not, if given, modifies the value in weight struct.
+	args.GetInt(&weight.selectorSpreadWeight, ImageLocalityWeight)
```
Comment (william-wang) [inline pkg/scheduler/plugins/nodeorder/nodeorder.go:152]: why use ImageLocalityWeight in `args.GetInt(&weight.selectorSpreadWeight, ImageLocalityWeight)`

Comment (elinx) [reply pkg/scheduler/plugins/nodeorder/nodeorder.go]: a copy-paste mistake 😿 , fixed.

File: pkg/scheduler/cache/cache.go [hunk 2]
```diff
@@ -640,6 +640,13 @@ func newSchedulerCache(config *rest.Config, schedulerNames []string, defaultQueu
 		UpdateFunc: sc.UpdateNumaInfoV1alpha1,
 		DeleteFunc: sc.DeleteNumaInfoV1alpha1,
 	})
+	// explictly register informers to the factory, otherwise resources listers cannot get anything
+	// even with no erorr returned.
+	informerFactory.Core().V1().Namespaces().Informer()
```
Comment (william-wang) [inline pkg/scheduler/cache/cache.go:645]: Only selectorSpread plugin and topology spread plugin need below informer register?

File: pkg/scheduler/cache/cache.go [hunk 3]
```diff
@@ -640,6 +640,13 @@ func newSchedulerCache(config *rest.Config, schedulerNames []string, defaultQueu
 		UpdateFunc: sc.UpdateNumaInfoV1alpha1,
 		DeleteFunc: sc.DeleteNumaInfoV1alpha1,
 	})
+	// explictly register informers to the factory, otherwise resources listers cannot get anything
```
Comment (william-wang) [inline pkg/scheduler/cache/cache.go:643]: suggest move these register to line 489 and add comments to declare that it is for selectorSpread plugin.

**Comment by william-wang** [comment]
@elinx please merge the two commits to one.

File: pkg/scheduler/cache/cache.go (hunk 2 above)
Comment (elinx) [reply pkg/scheduler/cache/cache.go:645]: The `InterPodAffinity` plugin need the `namespace` informer, `SelectorSpread` and `PodTopologySpread`  needs the following four.

File: pkg/scheduler/cache/cache.go (hunk 3 above)
Comment (elinx) [reply pkg/scheduler/cache/cache.go]: done

File: pkg/scheduler/cache/cache.go (hunk 2 above)
Comment (elinx) [reply pkg/scheduler/cache/cache.go]: > @elinx please merge the two commits to one.

done

**Review by william-wang — APPROVED** [review]
/lgtm

**Pull request merged**